web/src/settings_bots.js

```diff
@@ -416,27 +415,38 @@ export function show_edit_bot_info_modal(user_id, from_user_info_popover) {
 
     function edit_bot_post_render() {
         $("#edit_bot_modal .dialog_submit_button").prop("disabled", true);
-        const owner_id = bot_data.get(user_id).owner_id;
-
-        const user_ids = people.get_active_human_ids();
-        const users_list = user_ids.map((user_id) => ({
-            name: people.get_full_name(user_id),
-            value: user_id.toString(),
-        }));
-
-        const opts = {
-            widget_name: "edit_bot_owner",
-            data: users_list,
-            default_text: $t({defaultMessage: "No owner"}),
-            value: owner_id,
-            on_update(value) {
-                $("#edit_bot_modal .dialog_submit_button").prop("disabled", value === null);
-            },
-        };
+
+        function get_options() {
+            const user_ids = people.get_active_human_ids();
+            return user_ids.map((user_id) => ({
+                name: people.get_full_name(user_id),
+                unique_id: user_id,
+            }));
+        }
+
+        function item_click_callback(event, dropdown) {
+            const $user = $(event.currentTarget);
+            const user_full_name = $user.attr("data-name");
+            const new_bot_owner_id = Number.parseInt($user.attr("data-unique-id"), 10);
+            const $bot_owner = $("#bot_owner_dropdown_widget .bot_owner_name");
+            $bot_owner.text(user_full_name);
+            $bot_owner.attr("data-user-id", new_bot_owner_id);
+            $("#edit_bot_modal .dialog_submit_button").prop("disabled", user_full_name === null);
+            dropdown.hide();
+            event.stopPropagation();
+            event.preventDefault();
+        }
+
         // Note: Rendering this is quite expensive in
         // organizations with 10Ks of users.
```

Comment:
Is this comment still accurate? I wasn't sure whether the new code will actually do any work involving the list of options until the widget is actually opened -- if it doesn't, we may have fixed that bug?

Reply:
Yeah, removed the comment. Also, `ListWidget` makes sure we don't do much work rendering it.